frontend/src/metabase/public/LocaleProvider.tsx

```diff
@@ -3,27 +3,38 @@ import { type PropsWithChildren, useEffect, useState } from "react";
 import { setLocaleHeader } from "metabase/lib/api";
 import { loadLocalization, setUserLocale } from "metabase/lib/i18n";
 
+interface LocaleProviderProps {
+  locale?: string | null;
+  shouldWaitForLocale?: boolean;
+}
+
 export const LocaleProvider = ({
   children,
   locale,
-}: PropsWithChildren<{ locale?: string | null }>) => {
+  shouldWaitForLocale,
+}: PropsWithChildren<LocaleProviderProps>) => {
   // The state is not used explicitly, but we need to trigger a re-render when the locale changes
   // as changing the locale in ttag doesn't trigger react components to update
-  const [_isLoadingLocale, setIsLoadingLocale] = useState(false);
+  const [isLocaleLoaded, setIsLocaleLoaded] = useState(false);
 
   useEffect(() => {
     if (locale) {
-      setIsLoadingLocale(true);
       setLocaleHeader(locale);
       loadLocalization(locale).then(translatedObject => {
-        setIsLoadingLocale(false);
+        setIsLocaleLoaded(true);
         setUserLocale(translatedObject);
       });
     }
   }, [locale]);
 
-  // note: we may show a loader here while loading, this would prevent race
-  // conditions and things being rendered for some time with the wrong locale
-  // downside is that it would make the initial load slower
-  return <>{children}</>;
+  if (
```

Comment:
we want to render `children` immediately when:
1. we don't pass `locale` since there's nothing to load
2. we pass `locale`, but we don't want to block the rendering with so we don't pass `shouldWaitForLocale` or `shouldWaitForLocale: false`


In case when `shouldWaitForLocale: true`, we'll wait for `isLocaleLoaded: true` before showing rendering the children